src/core/umount.c

```diff
@@ -577,17 +646,19 @@ int umount_all(bool *changed) {
         bool umount_changed;
         int r;
 
-        /* retry umount, until nothing can be umounted anymore */
+        /* Retry umount, until nothing can be umounted anymore. Mounts are
+         * processed in order, newest first. The retries are needed when
+         * an old mount has been moved, to a path inside a newer mount. */
         do {
                 umount_changed = false;
 
-                umount_all_once(&umount_changed, false);
+                umount_all_once(&umount_changed);
                 if (umount_changed)
                         *changed = true;
         } while (umount_changed);
 
-        /* umount one more time with logging enabled */
-        r = umount_all_once(&umount_changed, true);
+        /* umount one more time just to be safe */
+        r = umount_all_once(&umount_changed);
```

Comment:
This served a specific purpose with the log_error flag, but it looks unjustified now.  It will also cause extra delays in the case we are working on here.

Reply:
True, and I like the direction. I can remove it with the next iteration.